client/core/trade.go

```diff
@@ -367,92 +382,98 @@ func (t *trackedTrade) negotiate(msgMatches []*msgjson.Match) error {
 			},
 		}
 
-		// First check that this isn't a match on its own cancel order. I'm not crazy
-		// about this, but I am detecting this case right now based on the Address
-		// field being an empty string.
-		isCancel := t.cancel != nil && msgMatch.Address == ""
-		if isCancel {
-			match.setStatus(order.MatchComplete)
```

Comment:
This `setStatus` must have been superfluous because the `match` for a cancel Match is not stored in the `t.matches` map or stored in the DB.  Should this cancel `msgMatch` get a MetaMatch stored in the DB too like the others?  The taker side of the match isn't stored either, so I think we aren't intending to store them, but I'm not sure.

Reply:
This is now addressed by storing both maker and taker Matches in https://github.com/decred/dcrdex/compare/2c42ccd942492b584e539076b36fc6b8c3c08699...c952eb9e8a9e0bfb9197bff9745e94b5b230884b#diff-ca928c370ec8b724badec3db0dbd7bc4R368
I think the taker's cancel Match not being stored was a bug.

Reply:
Thanks for this. This is the first step needed to fix https://github.com/decred/dcrdex/blob/be176bad73c8ffb765ecc118153bc690d448a5d9/client/core/core.go#L2303